math/module_argument_cplexnum.cpp

```diff
@@ -0,0 +1,52 @@
+/// c++ program to calculate the modulus and argument of a complex number
+#include <cmath>
+#include <iostream>
+/**
+ *Calculate the modulus and argument(radians) value for a complex number
+ *the form of a complex number be like z=x+iy
+ *so for a pair of numbers (a,b) that a represents x and b y
+ *the program can calculate the modulus like
+ *|z|=sqrt(a^2+b^2)
+ *and the argument like
+ *arg=arctan(y/x)
+ */
+struct  Numcpx {
+    double x , y;
+};
+
+/**
+ *The method use the Pythagorean theorem to calculate the modulus.
+ */
+double modulus(Numcpx a) {
+    double x_2 = std::pow(std::abs(a.x), 2);
+    double y_2 = std::pow(std::abs(a.y), 2);
+    return std::sqrt(x_2 + y_2 );
+}
+
+double argument(Numcpx a) {
+    double resulta = atan2(a.y , a.x);
+    return resulta;
+}
+
+void test() {
+/**
+*    for the pair numbers(4,3)
+*    the result must be 5 and argument be 0.6435
+*/
+    Numcpx numcpx = {4 , 3};
+    std::cout <<"modulus:"<< modulus(numcpx) << std::endl;
+    std::cout <<"argument:"<< argument(numcpx) << std::endl;
+}
+
+void test2() {
+    Numcpx numcpx = {4.6 , 3.3};
+    std::cout <<"modulus:"<< modulus(numcpx) << std::endl;
```

Comment:
test cases should be self-sufficient i.e., the test case should check for a known output using `assert` statements.
you can use the C++ standard `complex` library to check the output of your code with that by the library within the function itself. [Please see this code](https://github.com/TheAlgorithms/C-Plus-Plus/blob/55c21c86e1a916df446c0f983c19c81bec35dc49/math/fast_power.cpp#L62)